feat: ux updates

ypatil12 opened the pull request:
# v2.0.0 UX Improvements

This release brings 2 UX improvements to the middleware repo. We increment the major release due to the new `createSlashableStakeQuorum` interface. 

🚀 New Features

- NonSigner View Function for Bn254 Table Calculator: Constructs nonsigner witness onchain by passing in a list of signers

⛔ Breaking Changes
- Update `createSlashableStakeQuorum` to take in a slasher address

🔧 Improvements
- Move to foundry v1.5.0 and update formatting
- Upgrade solc to 0.8.29

## Changelog


- feat: update registry coordinator for new createOperatorSets  [PR #548](https://github.com/layr-labs/eigenlayer-middleware/pull/548)
- feat: nonsigner view and operator index [PR #545](https://github.com/Layr-Labs/eigenlayer-middleware/pull/542)
- chore: update readMe for middlewarev2 deployment [PR #539](https://github.com/layr-labs/eigenlayer-middleware/pull/539)

eigenmikem committed:
feat: nonsigner view and operator index

ypatil12 committed:
feat: update registry coordinator for new createOperatorSets  (#548)

**Motivation:**

As part of https://github.com/Layr-Labs/eigenlayer-contracts/pull/1645,
we are adding a new `createOperatorSets` function that passes in the
slasher address upon creation of an operatorSet. Although the *old*
`createOperatorSets` function is not being deprecated, we are updating
the `RegistryCoordinator` in order to be **only compatible the new
function**.

**Modifications:**

- Update `createSlashableStakeQuorum` to take in a `slasher` address
- Update all tests
- Update `solc` version in `foundry.toml` to `0.8.29`, matching Core

**Result:**

Compatible with updated `AllocationManager` interface

nadir-akhtar commented on src/SlashingRegistryCoordinator.sol:
```diff
@@ -125,7 +125,8 @@ contract SlashingRegistryCoordinator is
             IStakeRegistryTypes.StakeType.TOTAL_DELEGATED,
-            0
+            0,
+            DELEGATED_STAKE_SLASHER
```
Given that the `DELEGATED_STAKE_SLASHER` corresponds to the burn address and thus can never perform a slash, what's the reason for having no option to slash when creating an operator set via this function?

ypatil12 committed:
docs: v2.0.0 release notes; feat: rev contract submodule  (#550)

Release notes for v2.0.0. Also, rev version for core submodule.

ypatil12 committed:
docs: slashable stake quorum (#551)

**Motivation:**

Natspec and docs were missing changes to the
`createTotalDelegatedStakeQuorum` and `createTotalSlashableStakeQuorum`.

**Modifications:**

Update docs and natspec. 

**Result:**

Up to date docs.

---------

Co-authored-by: Nadir Akhtar <9601075+nadir-akhtar@users.noreply.github.com>

nadir-akhtar reviewed (approved):
The CI seems to be failing for a "storage report" failure in regards to the `RegistryCoordinator`, but manual examination shows that no material change has been done to it, meaning this warning is a false positive. @0xClandestine suspects this is due to differing Foundry issues and will debug locally.

That said, no need to block the PR on this false positive.
